contrib/tools/psbtsigner/src/main.rs

```diff
@@ -10,31 +10,37 @@ use revault_tx::{
     miniscript::descriptor::DescriptorSecretKey,
 };
 
-fn parse_args(args: Vec<String>) -> (PrivateKey, Psbt, usize) {
+fn parse_args(args: Vec<String>) -> (Vec<PrivateKey>, Psbt, usize) {
     if args.len() < 3 {
         eprintln!(
-            "Usage: '{} <xpriv/derivation/path OR wif_privkey> <psbt> [input index]'",
+            "Usage: '{} <xpriv/derivation/path OR wif_privkey> [<xpriv/derivation/path OR wif_privkey>, ...] <psbt> [input index]'",
             args[0]
         );
         process::exit(1);
     }
 
-    let key = DescriptorSecretKey::from_str(&args[1]).unwrap_or_else(|e| {
-        eprintln!("Error parsing private key: '{}'", e);
-        process::exit(1);
-    });
-    let key = match key {
-        DescriptorSecretKey::SinglePriv(single_priv) => single_priv.key,
-        DescriptorSecretKey::XPrv(xpriv) => {
-            let secp = secp256k1::Secp256k1::signing_only();
-            xpriv
-                .xkey
-                .derive_priv(&secp, &xpriv.derivation_path)
-                .unwrap()
-                .private_key
-        }
-    };
-    let psbt = base64::decode(&args[2]).unwrap_or_else(|e| {
+    let mut keys = vec![];
+    // Parsing keys until we find something else...
+    for i in 0.. {
```

Comment:
You should upperbound it so that `args[i + 1]` doesn't overflow? Starting at `1` would also make it more readable imo. something like
```rust
for i in 1..args.len() - 1 { }
```
[...]
```rust
DescriptorSecretKey::from_str(&args[i])
```